Update user-groups.php to avoid resaving usergroup for user already in group

jerclarke opened the pull request:
## Description

This small change represents an enormous performance improvement when there are a lot of users and groups. 

Without this check if the user is already in the group it gets re-saved anyway. If the user is in many groups this represents a lot of wasted time, and on our site it was slowing down user saving to the point where it felt broken. 

I see no risk in this check being added to avoid re-saving a user who is already in a group.  I've been running this as a hotfix in my plugin for years and saving users to groups works fine.

Thank you for considering it.

## Steps to Test

Outline the steps to test and verify the PR here.

* Add and remove users from groups
* Make sure everything works fine
* Enjoy the speedup caused by not resaving when you don't have to

GaryJones commented:
Hi @jerclarke, could you please rebase this PR onto `develop`? We've recently merged some CI fixes and want to get this tested and merged. Thanks\!

GaryJones committed:
Update user-groups.php to avoid resaving usergroup for user already in group

Skip updating the usergroup if the user is already a member, avoiding
unnecessary database writes.

Co-authored-by: Jer Clarke <jer@simianuprising.com>
